### PR TITLE
recovery: make initial CWND configurable

### PR DESCRIPTION
**Pull request by evanrittenhouse**

### CODE REVIEW

File: quiche/src/lib.rs [hunk 1]
```diff
@@ -1126,6 +1131,13 @@ impl Config {
         Ok(())
     }
 
+    /// Sets the number of packets to be sent in the initial congestion window.
```
Comment (LPardue) [inline quiche/src/lib.rs:1134]:
```suggestion
    /// Sets initial congestion window size in terms of packet count.
```

File: quiche/src/lib.rs [hunk 2]
```diff
@@ -483,6 +483,9 @@ const CONNECTION_WINDOW_FACTOR: f64 = 1.5;
 // validation as failed.
 const MAX_PROBING_TIMEOUTS: usize = 3;
 
+// The default number of packets sent in the initial congestion window.
```
Comment (LPardue) [inline quiche/src/lib.rs:486]:
```suggestion
// The default initial congestion window size in terms of packet count.
```

File: quiche/include/quiche.h [hunk 3]
```diff
@@ -212,6 +212,9 @@ void quiche_config_set_disable_active_migration(quiche_config *config, bool v);
 // Sets the congestion control algorithm used by string.
 int quiche_config_set_cc_algorithm_name(quiche_config *config, const char *algo);
 
+// Sets the initial cwnd for the connection.
```
Comment (LPardue) [inline quiche/include/quiche.h:215]:
```suggestion
// Sets the initial cwnd for the connection in terms of packet count.
```

File: quiche/src/recovery/mod.rs [hunk 4]
```diff
@@ -168,6 +165,9 @@ pub struct Recovery {
 
     /// How many non-ack-eliciting packets have been sent.
     outstanding_non_ack_eliciting: usize,
+
+    /// How many packets should be sent in the initial congestion window.
```
Comment (LPardue) [inline quiche/src/recovery/mod.rs:169]:
```suggestion
    /// Initial congestion window size in terms of packet count
```

**Review by LPardue — CHANGES_REQUESTED** [review]

**Comment by evanrittenhouse** [comment]
To test, I set `DEFAULT_INITIAL_CONGESTION_WINDOW_PACKETS` to 30 and checked QVIS on a 90 KB image. You can see here the difference between the two:

#### ICWND = 10
testing_icwnd_10

#### ICWND = 30
testing_icwnd_30

**Review by LPardue — APPROVED** [review]

**Comment by junhochoi** [comment]
👍 might be beneficial to add a configurable option to quiche-server/client for easier testing.

**Comment by evanrittenhouse** [comment]
Checked QVIS logs with `--initial-cwnd-packets 10` and `--initial-cwnd-packets 30` on both quiche-server and quiche-client. We see similar results as before:

**IWND = 10**
icwnd_10

**IWND = 30**
icwnd_30

**Review by ghedo — APPROVED** [review]

**Pull request merged**